frontend/src/components/workspace/DeleteSavedSearchModal.tsx

```diff
@@ -96,7 +96,7 @@ export function DeleteSavedSearchModal({
       .finally(() => {
         setLoading(false);
       });
-  }, [user?.token, replaceQueryParams, savedSearchId]);
+  }, [clientFetch, user?.token, replaceQueryParams, savedSearchId]);
```

Comment:
I had some trouble with infinite re-renders on some of these, which is why I didn't include the function in the dependency arrays to begin with. Are you sure we're not getting any infinite loops on these? Sometimes hard to tell other than slow performance unless you add a log for each render. Maybe something to consider splitting into a separate pr

Reply:
Of course!